pkg/operation/seed/seed.go

```diff
@@ -828,151 +828,351 @@ func BootstrapCluster(ctx context.Context, k8sGardenClient, k8sSeedClient kubern
-			tag        = version.Get().GitVersion
+			dnsProvider    = emptyDNSProvider()
+			providerSecret = emptyDNSProviderSecret()
 		)
-		if gardenerSeedAdmissionControllerImage.Tag != nil {
-			repository = gardenerSeedAdmissionControllerImage.Repository
-			tag = *gardenerSeedAdmissionControllerImage.Tag
-		}
-		gsacImage = imagevector.Image{
-			Repository: repository,
-			Tag:        &tag,
-		}
+
+		_, err := controllerutil.CreateOrUpdate(ctx, seedClient, dnsProvider, func() error {
+			dnsProvider.Spec = dnsv1alpha1.DNSProviderSpec{
+				Type: dnsConfig.Provider.Type,
+				SecretRef: &corev1.SecretReference{
+					Namespace: providerSecret.Namespace,
+					Name:      providerSecret.Name,
+				},
+			}
+
+			if dnsConfig.Provider.Domains != nil {
+				dnsProvider.Spec.Domains = &dnsv1alpha1.DNSSelection{
+					Include: dnsConfig.Provider.Domains.Include,
+					Exclude: dnsConfig.Provider.Domains.Exclude,
+				}
+			}
+
+			if dnsConfig.Provider.Zones != nil {
+				dnsProvider.Spec.Zones = &dnsv1alpha1.DNSSelection{
+					Include: dnsConfig.Provider.Zones.Include,
+					Exclude: dnsConfig.Provider.Zones.Exclude,
+				}
+			}
+
+			return nil
+		})
+		return err
 	}
-	components = append(components, seedadmissioncontroller.New(c.Client(), namespace, gsacImage.String(), kubernetesVersion))
+}
 
-	// kube-scheduler for shoot control plane pods
-	var schedulerImage *imagevector.Image
-	if imageVector != nil {
-		schedulerImage, err = imageVector.FindImage(charts.ImageNameKubeScheduler, imagevector.TargetVersion(kubernetesVersion.String()))
-		if err != nil {
-			return nil, err
-		}
+func destroyDNSProviderTask(seedClient client.Client, seedLogger *logrus.Entry) func(ctx context.Context) error {
+	return func(ctx context.Context) error {
+		return kutil.DeleteObjects(ctx, seedClient, emptyDNSProvider(), emptyDNSProviderSecret())
 	}
-	sched, err := gardenerkubescheduler.Bootstrap(c.DirectClient(), namespace, schedulerImage, kubernetesVersion)
-	if err != nil {
-		return nil, err
+}
+
+func emptyDNSProvider() *dnsv1alpha1.DNSProvider {
+	return &dnsv1alpha1.DNSProvider{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: v1beta1constants.GardenNamespace,
+			Name:      "seed",
+		},
 	}
-	components = append(components, sched)
+}
 
-	return components, nil
+func emptyDNSProviderSecret() *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: v1beta1constants.GardenNamespace,
+			Name:      "dnsprovider-seed",
+		},
+	}
 }
 
 func managedIngress(seed *Seed) bool {
 	return seed.Info.Spec.Ingress != nil
 }
 
+func getManagedIngressLoadBalancerAddress(ctx context.Context, k8sSeedClient kubernetes.Interface, seedLogger *logrus.Entry) (string, error) {
+	loadBalancerAddress, err := kutil.WaitUntilLoadBalancerIsReady(
+		ctx,
+		k8sSeedClient,
+		v1beta1constants.GardenNamespace,
+		"nginx-ingress-controller",
+		time.Minute,
+		seedLogger,
+	)
+	if err != nil {
+		return "", err
+	}
+	return loadBalancerAddress, nil
```

Comment:
```suggestion
	return kutil.WaitUntilLoadBalancerIsReady(
		ctx,
		k8sSeedClient,
		v1beta1constants.GardenNamespace,
		"nginx-ingress-controller",
		time.Minute,
		seedLogger,
	)
```

Generally, I'm wondering if this function is helpful or not (it's not encapsulating too much).

Reply:
Sometimes I use functions so I can give an operation a name in order to avoid writing a comment. In this case the function does not add any value - thanks.